apps/search/js/providers/suggestions.js

```diff
@@ -37,13 +37,15 @@
     },
 
     search: function(input) {
-      return new Promise((resolve, reject) => {
-        var url = encodeTerms(SearchProvider('suggestUrl'), input);
-        LazyLoader.getJSON(url, true).then(result => {
-          var results = result[1];
-          // We add an item to search the entered term as well
-          results.unshift(result[0]);
-          resolve(results);
+      return SearchProvider.ready().then(function() {
+        return new Promise((resolve, reject) => {
```

Comment:
I think you can use the ```resolve``` as the first argument from the ```.then()```? So basically standard promise chaining..

```
return SearchProvider.ready().then(resolve => {

});
```